### PR TITLE
Fix compatibility issues

### PR DESCRIPTION
**Pull request by fedebotu**
Here is a changelog of the fixes:
- Change `LQRStep` from `autograd.Functional` to `nn.Module` for compatibility (more info about legacy autograd function with non-static method [here](https://discuss.pytorch.org/t/legacy-autograd-function-with-non-static-forward-method-is-deprecated-and-will-be-removed-in-1-3/69869) and [here](https://github.com/NVIDIA/flownet2-pytorch/pull/207/commits/40b5bf9a220f58c3d0317204cf339028762b5789))
- Commented out `save_for_backward` (not needed in `nn.Module` apparently since it is using the Functional API for saving the weights by default, more info [here](https://discuss.pytorch.org/t/what-is-the-difference-between-torch-nn-and-torch-nn-functional/33597/7))
- `LQRStep` returns backward and forward trajectory by default
- Add local import in test files
- Add Circle CI testing (you can change the account info later)
- Miscellaneous fixes

### CODE REVIEW

**Comment by bamos** [comment]
Thanks for the updates! Are the gradient tests passing with this? I think replacing the Function with the Module will make it so the right backward pass doesn't get called ([here](https://github.com/locuslab/mpc.pytorch/pull/32/files#diff-cb8b433fbb077405aa9b56bca3ff45d033c684ba0ca74e94fff54ba70a0988f6R118)) and will make the derivatives silently wrong

**Comment by bamos** [comment]
Fixed in https://github.com/locuslab/mpc.pytorch/commit/bf19cd4405d8242e93f4c0a5464739f98798a833
